aten/src/ATen/core/dispatch/DispatchTable.h

```diff
@@ -247,6 +236,30 @@ class DispatchTable final {
     return {0, false, false};
   }
 
+  template<class GetDispatchKeyFunc>
+  const DispatchTableEntry& lookup_(const GetDispatchKeyFunc& getDispatchKey) const {
+    return kernels_.map<const DispatchTableEntry&>(
+      [&] (const detail::KernelTable_& table) -> const DispatchTableEntry& {
+        // We have a dispatch table. Find the correct kernel for the inputs and return it.
+
+        TORCH_INTERNAL_ASSERT(dispatch_strategy_.is_valid_, "Operator ", operator_name_, " has an invalid dispatch key but kernels registered.");
+
+        TensorTypeId dispatch_key = getDispatchKey();
```

Comment:
it's probably not a comment for now, but this code is pretty convoluted just to optimize the catch-all case, right? given that majority (all?) of our kernels would be using dispatch, we could have made this code more readable by folding all lambdas and just passing in the dispatch key directly.

Reply:
most custom ops don't use dispatch and internally, meta ops don't either.